Feature/2407：コーディングガイドライン改訂施策

depart-takizawa opened the pull request:
@depart-keigoyoshida
全体確認しまして、修正できそうな点についてコミットいたしました。
また、以下はコミット外で気になった点です。
ご確認よろしくお願いいたします。
---
### 動作環境
https://depart-develop.github.io/coding-guidelines/docs/guidelines/environment/
すぐ古くなってしまうので言及せずに曖昧にしたい。
WindowsはMSサポート対象とか、iOSは2version前まで、Androidはリリース5年前までシェア5%以上とかそういう定義の仕方。←適当です。。
（これはたぶん見積もり書や案件の対象環境も定型化され同様の問題が起こっていると思われるから意識的に統一する？）
### 行末尾の空白 / ファイル末尾の改行
https://depart-develop.github.io/coding-guidelines/docs/guidelines/general/#%E5%BF%85%E9%A0%88-%E8%A1%8C%E6%9C%AB%E5%B0%BE%E3%81%AE%E7%A9%BA%E7%99%BD
https://depart-develop.github.io/coding-guidelines/docs/guidelines/general/#%E5%BF%85%E9%A0%88-%E3%83%95%E3%82%A1%E3%82%A4%E3%83%AB%E6%9C%AB%E5%B0%BE%E3%81%AE%E6%94%B9%E8%A1%8C
どちらもよくわかりませんでした、、特にファイル末尾の改行は個人的に気になって削除していたりしたもので、、、差分検出のため…？
### 大文字は使用しない
https://depart-develop.github.io/coding-guidelines/docs/guidelines/css/#%E5%BF%85%E9%A0%88-%E5%A4%A7%E6%96%87%E5%AD%97%E3%81%AF%E4%BD%BF%E7%94%A8%E3%81%97%E3%81%AA%E3%81%84
削除か必須→推奨、もしくは例を変えたほうが良いかも。
HEXに関してはxdやfigmaからコピーすると基本大文字な気がしています（もしやコンパイル設定などで小文字になるようにしてたりしますか？）
### クラス名の記法 / CSS設計とその記法に関して
https://depart-develop.github.io/coding-guidelines/docs/naming_convention/css-name/
「ケバブケースを使用」と「BEM記法を採用」に矛盾を感じました。
### その他
CSS/JSなどは基本的に圧縮していることなどを鑑みると、全体としてのポリシーのようなものがあっても良い気がしました。
→自戒の意味と社内やクライアントの信頼にも繋がるかなと。。
参考：https://html-coding.co.jp/annex/guideline/guideline.php

depart-takizawa committed:
インデント
cssは基本圧縮するのかなと思うのであえて例に出さなくても良いかも？
https://depart-develop.github.io/coding-guidelines/docs/guidelines/general/

depart-takizawa committed:
クラス名の記法
後ページの「フレームワークでのコンポーネントに関して」でも言及されていますが、Next.jsでハイフン付きクラスのエラー回避記法が面倒でキャメルケースを多用したので（それでいてケバブケースもあったりする…）、振り幅を持たせたい。
https://depart-develop.github.io/coding-guidelines/docs/guidelines/css/

depart-takizawa committed:
カラーコードの16進表記
必須→推奨
「可能な限り」と記載があるので必須とは言い難いのと、xdやfigmaからコピーすると6桁なのを修正するのは手間なので。
https://depart-develop.github.io/coding-guidelines/docs/guidelines/css/

depart-takizawa committed:
ブロック内のインデント
必須→推奨
インデント項同様、基本圧縮するのでこの限りではないのかなと。

depart-takizawa committed:
セレクタおよび宣言の分離
例を同じものにしたほうが理解しやすい。
https://depart-develop.github.io/coding-guidelines/docs/guidelines/css/

depart-takizawa committed:
ルールの分離
必須→推奨
空白行を入れなくてはならない明確な理由が見当たらない…？（もちろん見通しを良くするはあるが、、）
https://depart-develop.github.io/coding-guidelines/docs/guidelines/css/

depart-takizawa committed:
演算子
「オペランド」の突然感
次項の「ブレース（波括弧）」のように「オペランド（被演算子）」とする？
https://depart-develop.github.io/coding-guidelines/docs/guidelines/js/

depart-takizawa committed:
セマンティクス
Goodのtableタグのインデント気になる
https://depart-develop.github.io/coding-guidelines/docs/guidelines/a11y/

depart-takizawa committed:
フォーム・コントロールのラベル
タグのインデント気になる、、というかフォームであればtableじゃないタグにすべきかも？
https://depart-develop.github.io/coding-guidelines/docs/guidelines/a11y/

depart-takizawa committed:
CSSクラス名リスト
「DEPARTルールとして」は削除で良い気がします。参考くらいのイメージ
https://depart-develop.github.io/coding-guidelines/docs/naming_convention/css-name/

depart-takizawa committed:
フレームワークでのコンポーネントに関して
タグのスペースがないのが気になる
https://depart-develop.github.io/coding-guidelines/docs/guidelines/css/#%E5%BF%85%E9%A0%88-%E5%AE%A3%E8%A8%8[…]F%E3%81%AE%E5%8C%BA%E5%88%87%E3%82%8A との矛盾
https://depart-develop.github.io/coding-guidelines/docs/naming_convention/css-sample-name/

depart-keigoyoshida reviewed (commented):
@depart-takizawa 
https://github.com/depart-develop/coding-guidelines/pull/3/commits/5a6719920f7341de3e474165fff2c5bf4344c461

> フォーム・コントロールのラベル
タグのインデント気になる、、というかフォームであればtableじゃないタグにすべきかも？
https://depart-develop.github.io/coding-guidelines/docs/guidelines/a11y/

インデントされていないのは確かに…NGですね！修正ありがとうございます🙇
table要素にしているのは、label要素でinput要素を囲えないパターン（要素が離れているので）として提示するためでした！
なので、table要素はそのままとさせてください！

depart-keigoyoshida commented:
@depart-takizawa 
https://github.com/depart-develop/coding-guidelines/pull/3/commits/ce684cb7c521f3f8776ad1aed858b255b1522ca3

> ブロック内のインデント
必須→推奨
インデント項同様、基本圧縮するのでこの限りではないのかなと。

圧縮する前のコードに対するスタイルルールでした！
なので、インデントありは必須とさせてください🙇

depart-keigoyoshida commented:
@depart-takizawa 
https://github.com/depart-develop/coding-guidelines/pull/3/commits/012dae5f6a34e8d316889d33064a51fe49f63eac

> カラーコードの16進表記
必須→推奨
「可能な限り」と記載があるので必須とは言い難いのと、xdやfigmaからコピーすると6桁なのを修正するのは手間なので。
https://depart-develop.github.io/coding-guidelines/docs/guidelines/css/

「可能な限り」というのは、3桁で表記できないカラーコードがあるからだと思いますが
プロジェクト内で書き方が統一されていれば問題ありませんので、「推奨」ルールにします！
ご意見ありがとうございます！

depart-keigoyoshida commented:
@depart-takizawa 
https://github.com/depart-develop/coding-guidelines/pull/3/commits/c6eecb7f2fa6c3b8579a72c2168004e14d8b8afe

> クラス名の記法
後ページの「フレームワークでのコンポーネントに関して」でも言及されていますが、Next.jsでハイフン付きクラスのエラー回避記法が面倒でキャメルケースを多用したので（それでいてケバブケースもあったりする…）、振り幅を持たせたい。
https://depart-develop.github.io/coding-guidelines/docs/guidelines/css/

確かに、使用するライブラリやフレームワークによってはキャメルケースで書いたほうがよいですね。。
プロジェクトごとにルールを検討するのは、この項目に限らずガイドライン全体に言えることですので
下記の文言だといかがでしょうか？

```
基本的に、単語の連結記法はハイフン区切りのケバブケースを使用する。
ただし、ライブラリ等の制約でケバブケース以外の記法ルールが適している場合はこの限りではない。
```

depart-keigoyoshida commented:
@depart-takizawa 
https://github.com/depart-develop/coding-guidelines/pull/3/commits/73f4a82708ddc9a5bbded56dfa41be63c998c66d

> インデント
cssは基本圧縮するのかなと思うのであえて例に出さなくても良いかも？
https://depart-develop.github.io/coding-guidelines/docs/guidelines/general/

便宜上CSSと書いていますが、SCSSをコーディングする際も同様で
「圧縮する前の手元で触るコードとして、どういったスタイルにするか」のルールになりますので、このままとさせてください！
（SCSSもタブでインデントしてほしくないという願い）

depart-keigoyoshida commented:
@depart-takizawa 
> 動作環境
https://depart-develop.github.io/coding-guidelines/docs/guidelines/environment/
すぐ古くなってしまうので言及せずに曖昧にしたい。

ご指摘ありがとうございます！ごもっともですね…！
むしろ、必ずプロジェクトごとに策定するものだと思いますので
このコーディングガイドラインには記載しないほうがよいかもしれないと思いました！
もうこのページ自体削除しますか。

depart-keigoyoshida commented:
@depart-takizawa 
> 行末尾の空白 / ファイル末尾の改行

ファイル末尾にコードを追加した際、差分がわかりにくくなってしまうのですよね。。
理由は、下記に詳しく記載がありました。

なぜファイルの末尾に改行を入れたほうが良いのか
https://qiita.com/hamacccccchan/items/11c17c7412a5aeb2ad74

社内の開発環境テンプレートでもファイル末尾に改行を自動追加する設定になっていますので
可能であれば、このままとさせてください！

depart-keigoyoshida commented:
@depart-takizawa 
> 大文字は使用しない

プロジェクトごとに統一されていれば問題ないと思いますので
（統一されていなければ、一括置換などの弊害になっちゃいますが）
ご提案いただいた通り、「必須」→「推奨」へ変更で問題ありません！

depart-takizawa committed:
[restore] インデント

depart-takizawa committed:
[restore] ブロック内のインデント

depart-takizawa committed:
[restore] フォーム・コントロールのラベル

depart-takizawa committed:
[mod] 大文字は使用しない
プロジェクトごとに統一されていれば問題ないと思いますので
（統一されていなければ、一括置換などの弊害になっちゃいますが）
ご提案いただいた通り、「必須」→「推奨」へ変更で問題ありません！

depart-takizawa committed:
クラス名の記法
ご提案いただいた文言に差し替え＋例外について記載（単独で読んだときにアンダースコア自体を使ってはいけないと認識してしまう可能性があるため）

depart-takizawa commented:
こちらのプルリクはクローズし、改めて作成します。
